### ISSUE TITLE
Quality of Simplify-X fits for H36M

### ISSUE THREAD

**Issue by iqbalu** [mks0601/I2L-MeshNet_RELEASE#22]
Hi @mks0601, 

Thanks a lot for sharing the Simplify-X fits for H36M. I was checking the quality of fits and it seems quite off when overlayed on the images. 

Here is an example: 

![Capture](https://user-images.githubusercontent.com/5772474/94940602-8f3af080-0488-11eb-98d2-beecb2f33f7a.JPG)

Is it what you were also getting or am I doing something wrong? 

Best,
Umar

**Comment by mks0601** [comment]
Hi Umar,

First of all, let me show the fitting errors at below table, which is from my paper.
I think 13 mm error is reasonable to be used as pseudo-GT.

![스크린샷 2020-10-03 오전 1 12 29](https://user-images.githubusercontent.com/13601723/94946156-ac1e0500-0516-11eb-9045-986b59e23c79.png)


It is hard to compare the fitting errors with those of Mosh fits (previously used ones by papers) because I haven't got access to the Mosh fits due to the license issue.
I downloaded the (maybe) Mosh fits from somewhere and remember that the errors of the Mosh fits were about 20~25 mm. I'm not perfectly sure the downloaded one is previously used one :(

Could you tell me how did you obtain the camera-centered 3D coordinates of 2D-projected coordinates from my fits?
Although there are some fitting errors, I think the overlayed results seem not good based on my experience.
The image-overlayed fits can look bad when you perform some data augmentations.

**Comment by iqbalu** [comment]
Hi @mks0601, 

Thanks a lot for the prompt reply. This is what I do, I basically copied the code from your data loader. I also downloaded the images that you have downloaded from google-drive. 

```
root_joint_idx = 0
root_pose = smpl_pose[root_joint_idx, :]
root_pose, _ = cv.Rodrigues(root_pose)
root_pose, _ = cv.Rodrigues(np.dot(R, root_pose))
smpl_pose[root_joint_idx] = root_pose.reshape(1,3)
output = smpl['neutral'](betas=torch.from_numpy(smpl_shape).unsqueeze(0).float(),
                                       body_pose=torch.from_numpy(smpl_pose[1:]).unsqueeze(0).float(),
                                       global_orient=torch.from_numpy(smpl_pose[0:1]).unsqueeze(0).float(),
                                      )
smpl_joint_coord = output.joints.squeeze().numpy()
smpl_mesh_coord = output.vertices.squeeze().numpy()

smpl_trans = np.dot(R, smpl_trans.T).reshape(1, 3) + T.reshape(1, 3) / 1000
root_joint_coord = smpl_joint_coord[root_joint_idx].reshape(1, 3)
smpl_trans = smpl_trans - root_joint_coord + np.dot(R, root_joint_coord.transpose(1,0)).transpose(1, 0)
smpl_mesh_coord = smpl_mesh_coord + smpl_trans
```

**Comment by mks0601** [comment]
It turns out that he was using a different joint regressor from that of SMPL.

**Comment by zhihaolee** [comment]
Hi @mks0601 ，

thank you for sharing the pseudo GT. 

There is a fitting threshold in your code to check if the fit is valid, 25 mm. Could you tell me, how many fits is invalid in the pseudo GT? Thanks!

Best,
Joey

**Comment by ShirleyMaxx** [comment]
Hi, @mks0601,
    Sorry to bother you. I've computed the fitting errors of your provided SMPLify-X fits, on the training set, the error is 25.72mm, and on the test set, the error is 31.24mm, instead of 13.1mm. Could you please help me figure out the reason?
    
1. gt pose3d is from `joint_cam`

2. fitted pose3d is computed by using the `joint_regressor_h36m `with the `mech `coordinates computed by the function `get_smpl_coord `using `shape`, `pose`, and `trans `in your code,

3. I've aligned both to their root. 
    
It seemed that I didn't miss anything? So is it available for you to provide your evaluation code of which get 13.1mm?
    Thank you in advance!

Best,
Shirley
